src/core/namespace.c

```diff
@@ -469,6 +469,32 @@ static void drop_outside_root(const char *root_directory, MountEntry *m, unsigne
         *n = t - m;
 }
 
+static void mount_mkdir_apivfs(const char *root_directory, MountEntry *m)
+{
+        const char *p, *path;
+
+        assert(m);
+
+        /*
+         * In case RootDirectory= or RootImage= are set, create the /proc, /sys
+         * and /dev directories, it is more likely to be missing when the
+         * service is started under these options. This is consistent with
+         * mount units creating the mount points when missing.
+         */
+        if (!root_directory)
+                return;
+
+        path = mount_entry_path(m);
+        p = startswith(path, root_directory);
+
+        /*
+         * Compare by paths instead of modes since we may bind mount or make
+         * paths READONLY which may hide the original/default mount mode
+         */
+        if (PATH_IN_SET(p, "/proc", "/dev", "/sys"))
```

Comment:
With m->mode is definitively better but I thought that this could be hidden or dropped due to mode  comparison and overwrite where we do BIND_MOUNT, PROCFS or READWRITE mode on the same /proc path ? will check again